src/python/pants/backend/python/lint/isort/rules.py

```diff
@@ -57,10 +63,48 @@ class Setup:
     original_digest: Digest
 
 
+def is_old_isort_version(version: Optional[str]) -> bool:
+    if version is None:
+        return False
+
+    requirement = cast(PackagingRequirement, Requirement.parse(version))
+    specifier = requirement.specifier
+
+    old_releases = (
+        *[f"4.3.{i}" for i in range(22)],
+        *[f"4.2.{i}" for i in range(16)],
+        *[f"4.1.{i}" for i in range(3)],
+        "4.0.0",
+    )
+    old_releases_match = any(old_release in specifier for old_release in old_releases)
+    if not old_releases_match:
+        return False  # if no old release matches, the version must be new
+
+    # if old releases match, the version may still be >=5 (e.g. isort!=5.0.0 or simply isort)
+    new_releases_match = any(  # This will match as soon as a new version matches
+        f"{major}.{minor}.{patch}" in specifier
+        for major in range(5, 10)
+        for minor in range(20)
+        for patch in range(50)
+    )
+    return not new_releases_match
+
+
 def generate_args(*, source_files: SourceFiles, isort: Isort, check_only: bool) -> Tuple[str, ...]:
-    # NB: isort auto-discovers config files. There is no way to hardcode them via command line
-    # flags. So long as the files are in the Pex's input files, isort will use the config.
+    # NB: isort auto-discovers config files in versions <5. Only for isort>=5 we need to hardcode them via command line
+    # flags.
```

Comment:
This is misleading. Isort always auto-discovers config files. In isort>5, we can also hardcode them via the CLI flag.

Reply:
agreed